public/components/visualization/embeddable/nlq_vis_embeddable.ts

```diff
@@ -93,12 +98,27 @@ export class NLQVisualizationEmbeddable extends Embeddable<
     return pipeline;
   };
 
+  private handleChange() {
```

Comment:
Nit: Seems like a dirtyChecker, the name is kind of similar to updateHandler.

Reply:
Perhaps we can name it dirtyCheck, and move the code that calls `this.updateHandler()` to subscription.

Reply:
sounds good, I will update it